## generated by make_metadata_documentation(): do not edit by hand 
## see source in data-raw/make_metadata_documentation.R 
#' Combined metadata of all studies in curatedMetagenomicData
#' 
#' The combined sample level metadata of all studies in curatedMetagenomicData.
#' 
#' @format A data.frame with 5716 rows and 80 variables:
#' \describe{
#'   \item{NA}{NA}
#'   \item{sampleID}{Sample identifier.}
#'   \item{subjectID}{Subject identifier.}
#'   \item{body_site}{Bodysite of acquisition.}
#'   \item{antibiotics_current_use}{Subject is currently taking antibiotics.}
#'   \item{study_condition}{The main disease or condition under study; control for controls.}
#'   \item{disease}{Semicolon-delimited vector of conditions; Use healthy only if subject is known to be healthy; CRC=colorectal cancer.}
#'   \item{age}{Subject age (years).}
#'   \item{infant_age}{Infant age (days); should be used for infants < 2 years old.}
#'   \item{age_category}{Age category: newborn < 1 year; 1 <= child < 12; 12 <= schoolage < 19; 19 <= adult <= 65; senior > 65.}
#'   \item{gender}{Subject gender.}
#'   \item{country}{Country of acquisition using ISO3 code from http://www.fao.org/countryprofiles/iso3list/en/.}
#'   \item{non_westernized}{Subject belongs to a non-westernized community.}
#'   \item{sequencing_platform}{This will be modified as new sequencing platforms are added to the database.}
#'   \item{DNA_extraction_kit}{DNA extraction kit.}
#'   \item{PMID}{Identifier of the main publication in PubMed.}
#'   \item{number_reads}{Number of final reads - calculated from raw data.}
#'   \item{number_bases}{Total number of bases sequenced in the sample.}
#'   \item{minimum_read_length}{Minimum read length - calculated from raw data.}
#'   \item{median_read_length}{Median read length - calculated from raw data.}
#'   \item{pregnant}{Pregnancy of the subject (men: no).}
#'   \item{lactating}{Lactating subjects (men: no).}
#'   \item{NCBI_accession}{Semicolon-separated vector of NCBI accessions.}
#'   \item{BMI}{Body mass index (kg/m2).}
#'   \item{antibiotics_family}{Family of antibiotics currently used; Semicolon-separated.}
#'   \item{momeducat}{Years of education of the mother of the subject.}
#'   \item{alcohol}{Subject is reported as a drinker.}
#'   \item{flg-genotype}{Any term for filaggrin-protein genotype.}
#'   \item{disease_subtype}{Disease subtype; CD=Chrohn's Disease.}
#'   \item{hdl}{Curators must use mg/l.}
#'   \item{triglycerides}{Curators must use mg/l.}
#'   \item{hba1c}{Curators must use \%.}
#'   \item{ldl}{Curators must use mg/l.}
#'   \item{tnm}{TNM classification for colorectal-cancer.}
#'   \item{body_subsite}{Subsite of body site of acquisition.}
#'   \item{visit_number}{Visit number for studies with repeated visits.}
#'   \item{days_from_first_collection}{Used for time series studies.}
#'   \item{c-peptide}{Curators must use ng/ml.}
#'   \item{family}{A number identifying the family subjects belong; not corrected for meta-analyses.}
#'   \item{cholesterol}{Curators must use mg/dl.}
#'   \item{glucose}{Curators must use mg/dl.}
#'   \item{mumps}{Subject has been through mumps in life.}
#'   \item{adiponectin}{Curators must use mg/l.}
#'   \item{insulin(cat)}{Insulin intake as a boolean.}
#'   \item{fgf-19}{Curators must use pg/ml.}
#'   \item{hscrp}{.}
#'   \item{leptin}{Curators must use micrograms/l.}
#'   \item{glutamate_decarboxylase_2_antibody}{.}
#'   \item{creatinine}{Curators must use micro-mol/l.}
#'   \item{il-1}{Curators must use pg/ml.}
#'   \item{cd163}{Curators must use ng/ml.}
#'   \item{glp-1}{Curators must use pmol/l.}
#'   \item{hitchip_probe_class}{High/Low species content onthe HIT-chip probe.}
#'   \item{hitchip_probe_number}{HIT-chip probe score.}
#'   \item{protein_intake}{Indication about the protein intake in the Mongolians diet.}
#'   \item{days_after_onset}{Days from the onset of the disease.}
#'   \item{stec_count}{.}
#'   \item{shigatoxin_2_elisa}{.}
#'   \item{stool_texture}{Texture of the stool at sampling time.}
#'   \item{ferm_milk_prod_consumer}{Dfmp means yes (defined milk product).}
#'   \item{mgs_richness}{Metagenomic species richness.}
#'   \item{location}{Free-form additional location information.}
#'   \item{dyastolic_p}{Measured in mm/Hg.}
#'   \item{systolic_p}{Measured in mm/Hg.}
#'   \item{prothrombin_time}{Prothrombin time in seconds.}
#'   \item{creatine}{Curators must use micro-mol/l.}
#'   \item{inr}{International normalized ratio.}
#'   \item{ctp}{.}
#'   \item{albumine}{Albumine level; curators must use g/l.}
#'   \item{bilubirin}{Bilubirin; curators must use mg/dl.}
#'   \item{smoker}{Currently a smoker at sampling.}
#'   \item{ever_smoker}{Ever been a smoker.}
#'   \item{birth_control_pil}{Use of the birth-control-pils at the sampling time (men: no).}
#'   \item{hla_drb12}{Hla_drb12 allele.}
#'   \item{hla_dqa12}{Hla_dqa12 allele.}
#'   \item{hla_dqa11}{Hla_dqa11 allele.}
#'   \item{hla_drb11}{Hla_drb11 allele.}
#'   \item{start_solidfood}{First day of solid food introduction (newborns).}
#'   \item{ajcc}{AJCC staging for colorectal-cancer.}
#'   \item{fobt}{Fecal occult blood test.}
#' }
#' 
#' @source See dataset specific help functions for source information 
"combined_metadata"
